spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -295,6 +321,18 @@ public Subscriber createSubscriber(String subscriptionName, MessageReceiver rece
       subscriberBuilder.setMaxAckExtensionPeriod(ackExtensionPeriod);
     }
 
+    Duration durationPerAckExtension = getMinDurationPerAckExtension(subscriptionName);
+    if (durationPerAckExtension != null
+        && !durationPerAckExtension.equals(DEFAULT_MIN_DURATION_PER_ACK_EXTENSION)) {
```

Comment:
There is a break here if `!durationPerAckExtension.equals(DEFAULT_MIN_DURATION_PER_ACK_EXTENSION))` is removed.

We can set `minDurationPerAckExtension` to `0` first, which basically do nothing but the server side will deem the default value is not used. 

If we also want to set `maxDurationPerAckExtension` to `0`, then the validation logic requires `minDurationPerAckExtension` is strictly smaller than `maxDurationPerAckExtension` if the default value of `minDurationPerAckExtension` is not used. An illegal state exception will throw in this case.

You can check `setMinDurationPerAckExtension` method in [here](https://github.com/googleapis/java-pubsub/blob/main/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java).

Adding the logic allows the user set both value to `0`. Though I guess few will do that, they will feel surprised if the application breaks since both of them set to the default value.

Reply:
Great investigation! I did not realize there was so much [additional logic](https://github.com/googleapis/java-pubsub/blob/a19bc7a6bd54a9223575c23df1cac7b2583eb61a/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java#L624) to determining if the default value was used. Would you be able to add a small comment for the condition mentioning this?

Reply:
Sure, I'll add comments in later commits.